src/Orchard.Cms.Web/Modules/Orchard.Roles/Services/RoleStore.cs

```diff
@@ -58,10 +58,10 @@ public void UpdateRoles(RolesDocument roles)
             _memoryCache.Set(Key, roles);
         }
 
-        public async Task<IEnumerable<string>> GetRoleNamesAsync()
+        public async Task<IEnumerable<RoleNamesEntry>> GetRoleNamesAsync()
```

Comment:
@sebastienros I did this change because RoleStore only allows to retrieve a NormalizedName, but not by Name. But I need both, Name for showing it to the user and the other to be able of retrieving a Role per NormalizedName to get its Claims in AccessController